src/libraries/Microsoft.Extensions.Logging/gen/LoggerMessageGenerator.Parser.cs

```diff
@@ -0,0 +1,661 @@
+                                            keepMethod = false;
+                                        }
+
+                                        if (method.Body != null)
+                                        {
+                                            Diag(DiagnosticDescriptors.LoggingMethodHasBody, method.Body.GetLocation());
+                                            keepMethod = false;
+                                        }
+
+                                        // ensure there are no duplicate ids.
+                                        if (ids.Contains(lm.EventId))
+                                        {
+                                            Diag(DiagnosticDescriptors.ShouldntReuseEventIds, ma.GetLocation(), lm.EventId, classDec.Identifier.Text);
+                                        }
+                                        else
+                                        {
+                                            _ = ids.Add(lm.EventId);
+                                        }
+
+                                        string msg = lm.Message;
+                                        if (msg.StartsWith("INFORMATION:", StringComparison.OrdinalIgnoreCase)
+                                            || msg.StartsWith("INFO:", StringComparison.OrdinalIgnoreCase)
+                                            || msg.StartsWith("WARNING:", StringComparison.OrdinalIgnoreCase)
+                                            || msg.StartsWith("WARN:", StringComparison.OrdinalIgnoreCase)
+                                            || msg.StartsWith("ERROR:", StringComparison.OrdinalIgnoreCase)
+                                            || msg.StartsWith("ERR:", StringComparison.OrdinalIgnoreCase))
+                                        {
+                                            Diag(DiagnosticDescriptors.RedundantQualifierInMessage, ma.GetLocation(), method.Identifier.ToString());
+                                        }
+
+                                        bool foundLogger = false;
+                                        bool foundException = false;
+                                        bool foundLogLevel = level != null;
+                                        foreach (ParameterSyntax p in method.ParameterList.Parameters)
+                                        {
+                                            string paramName = p.Identifier.ToString();
+                                            if (string.IsNullOrWhiteSpace(paramName))
+                                            {
+                                                // semantic problem, just bail quietly
+                                                keepMethod = false;
+                                                break;
+                                            }
+
+                                            IParameterSymbol? declSymbol = sm.GetDeclaredSymbol(p);
```

Comment:
Instead of having to go through the semantic model again, could you use the `methodSymbol` to get it's `IParameterSymbol`? It seems like you could use: https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.imethodsymbol.parameters?view=roslyn-dotnet#Microsoft_CodeAnalysis_IMethodSymbol_Parameters

Reply:
Feel free to address on a follow up PR.